### ISSUE TITLE
Monorepo, References: Output file has not been built from source file (TS6305)

### ISSUE THREAD

**Issue by xnerhu** [TypeStrong/fork-ts-checker-webpack-plugin#525]
## Current behavior
It throws 
```bash
TS6305: Output file 'C:/Users/xnerhu/Desktop/fork-ts-checker-webpack-plugin-repro/packages/foo/src/index.d.ts' has
not been built from source file 'C:/Users/xnerhu/Desktop/fork-ts-checker-webpack-plugin-repro/packages/foo/src/index.ts'
````

## Expected behavior
It should not throw the error.

## Steps to reproduce the issue
Import a package with typings referencing source files only (not prebuilt declarations). In VSCode it works.

### Example:

package/foo/index.ts
```ts
import * as test from 'bar'
```

package/bar/package.json
```json
{
  "name": "bar",
  "main": "build/main.js",
  "types": "src",
}
```

## Issue reproduction repository
[https://github.com/xnerhu/fork-ts-checker-webpack-plugin-repro](https://github.com/xnerhu/fork-ts-checker-webpack-plugin-repro)

## Environment
 - **fork-ts-checker-webpack-plugin**: ^5.2.1
 - **typescript**: ^4.0.5
 - **eslint**: none
 - **webpack**: ^5.3.2
 - **os**: Windows 10

**Comment by piotr-oles** [comment]
Hi! Did you tried `build: true` option?

**Comment by xnerhu** [comment]
`build: true` makes using the webpack pointless. The problem is that there are no built declaration for `bar`, so it needs to use the source code itself for the declarations, but unfortunately it doesn't work that way. I solved it by disabling `transpileOnly` in ts-loader.

**Comment by piotr-oles** [comment]
Could you elaborate on why `build: true` makes using the webpack pointless?

**Comment by felixmosh** [comment]
Having the same setup, monorepo with "common" package, and "project" package which consumes the "common".

**Comment by freshollie** [comment]
This is still an issue with `typescript@4.1.2`: https://github.com/freshollie/fresh-configurator/pull/8/checks?check_run_id=1560811910

**Comment by freshollie** [comment]
> Hi! Did you tried build: true option?

@piotr-oles ~~FYI this makes no difference~~

Edit: Turns out I had not upgraded to the latest version so didn't have this option available, solved my issue

**Comment by casamia918** [comment]
## Tips to create-react-app users.

At the beginning, I've been using CRA version 4.0.3.

In create-react-app version 4.0.3, they internally use fork-ts-checker-webpack-plugin version 4.1.6, which has no options of `build: true`. 
(see [package.json](https://github.com/facebook/create-react-app/blob/v4.0.3/packages/react-dev-utils/package.json#L64) )

I tried to manually set fork-ts-checker-webpack-plugin version to 5 by using package.json > resolutions field. But as the fork-ts plugin's option signature is changed, it failed with compile error. 

So, I have no choice without upgrading cra v5.

(You may be facing unexpected error while  upgrading cra v5. This is up to you. In my case, I've faced only 2 errors. )

fork-ts-checker-webpack-plugin introduced `options.typescript.build` at version 5

In create-react-app version 5, fork-ts-checker-webpack-plugin version is upgrade to ^6.5.0, so you can use build option.
(see [package.json](https://github.com/facebook/create-react-app/blob/v5.0.0/packages/react-dev-utils/package.json#L64) )

But CRA team did not set `build:true` option. So, if you are not ejecting cra, you should manually override the fork-ts-checker-webpack-plugin option.

In my case, I'm using react-app-rewired & customize-cra to override webpack config of CRA without ejecting.

So I add tiny overrideFunc to set build:true manually.

```js

module.exports = {
 webpack: override(
 ...
  (config) => {
    const forkTsPlugInInstances = config.plugins.find(
      (p) => p.constructor.name === 'ForkTsCheckerWebpackPlugin',
    );
    if (!forkTsPlugInInstances) return config;

    forkTsPlugInInstances.options.typescript.build = true;

    return config;
  },
 ...
)

}

```
